componentified-todo-list/prdaneshi/to-do-list/src/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const App = () => {
+  let [items, setItem] = useState([
+    { task: "first element", status: false },
+    { task: "second element", status: true },
+    { task: "third element", status: false },
+  ]);
+
+  const addToBotton = (event) => {
+    const newTask = {
+      task: inputValue,
+      status: false,
+    };
+
+    setItem([...items, newTask]);
+  };
+
+  const addToTop = (event) => {
+    const newTask = {
+      task: inputValue,
+      status: false,
+    };
+
+    setItem([newTask, ...items]);
+  };
+
+  const checkBoxClicked = (e, index) => {
+    console.log(index);
+    items[index].status = e.target.checked;
+    if (e.target.checked) {
+      const checked = items.splice(index, 1);
+      items.push(checked[0]);
+    }
+    setItem([...items]);
+  };
+
+  const DeleteTask = (e, index) => {
+    items.splice(index, 1);
+    setItem([...items]);
+  };
+
+  const [inputValue, setinputValue] = useState(0);
+  const InputValue = (e) => {
+    setinputValue(e.target.value);
+  };
+
+  const CreatList = () => {
+    const result = Object.entries(items).map((item) => (
+      <List
+        id={item[0]}
```

Comment:
__[MEDIUM]__  
Using array indexes as `id` is not good at all and causes bugs. Try to generate unique ids for each list item. This is an example on how you can generate an id:
```js
const uniqueId = Date.now();
```